# **Bootcamp Data Science & Analítica de Datos: Caso TITANIC**

<figure>
<center>
<img src='https://upload.wikimedia.org/wikipedia/commons/thumb/f/fd/RMS_Titanic_3.jpg/1920px-RMS_Titanic_3.jpg' width="650" height="250" />
<figcaption></figcaption></center>
</figure>

En este Bootcamp, aprenderemos sobre ciencia de datos a través de un ejemplo en el que exploraremos cómo la inteligencia artificial y la analítica de datos podrían haber sido de ayuda.

### Importamos librerias

In [11]:
# Importamos librerias para procesamiento de datos y gráficas
import os
import pandas as pd
import numpy as np
import seaborn as sns
import warnings
import matplotlib.pyplot as plt
import squarify
%matplotlib inline

plt.rcParams["figure.figsize"] = [10, 5]

# Importamos librerias para modelos de clasificación
from funciones_bootcamp import compute_metrics, plot_confusion_matrix_and_reports
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier, plot_tree

pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

# **Capítulo 1: Variables y Análisis**

## **1.1 Dataset, variables y tipos de variables**

### Carga del Dataset

Con el siguiente comando podemos extraer información almacenada en archivos csv.

In [12]:
# Se define la ruta del archivo .csv del cuál se quiere extraer información
ruta = '../Data/Inputs/Titanic-Dataset.csv'

# Se usa la función read_csv para cargar datos del archivo en la ruta especificada
full_data = pd.read_csv(ruta)

Con el siguiente comando podemos obtener las dimensiones de nuestro dataset

In [13]:
# Se usa la función shape de pandas para extraer el número de filas y columnas (filas, columnas)
full_data.shape

(891, 12)

El siguiente comando nos permitirá tener un acercamiento al dataset, observando los 10 primeros registros

In [14]:
# Se usa la función head para desplegar 10 registros del dataset
full_data.head(10)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
5,6,0,3,"Moran, Mr. James",male,NaN,0,0,330877,8.4583,NaN,Q
6,7,0,1,"McCarthy, Mr. Timothy J",male,54.0,0,0,17463,51.8625,E46,S
7,8,0,3,"Palsson, Master. Gosta Leonard",male,2.0,3,1,349909,21.0750,NaN,S
8,9,1,3,"Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)",female,27.0,0,2,347742,11.1333,NaN,S
9,10,1,2,"Nasser, Mrs. Nicholas (Adele Achem)",female,14.0,1,0,237736,30.0708,NaN,C


### Tipos de Variables

## Tipos de Variables en Análisis de Datos

En el análisis de datos, las **variables** son las características que describen a los elementos de un conjunto de datos. Estas variables pueden ser de diferentes tipos, y su clasificación es fundamental para elegir las técnicas estadísticas adecuadas para analizar la información.

### Clasificación de las Variables

Las variables se pueden clasificar en dos grandes grupos:

### 1. Variables Cuantitativas
Representan cantidades numéricas y pueden ser medidas.

- **Discretas**: Toman valores enteros y representan cantidades que se pueden contar.
  **Ejemplos**: número de hijos, cantidad de carros en un estacionamiento, número de llamadas recibidas.

- **Continuas**: Pueden tomar cualquier valor dentro de un intervalo y representan medidas.
  **Ejemplos**: altura, peso, temperatura, tiempo.

### 2. Variables Cualitativas (Categóricas)
Expresan cualidades o características que no pueden ser medidas numéricamente.

- **Nominales**: No existe un orden natural entre las categorías.
  **Ejemplos**: color de cabello, género, país de origen.

- **Ordinales**: Existe un orden natural entre las categorías, pero no se pueden medir las distancias entre ellas.
  **Ejemplos**: nivel de estudios (primaria, secundaria, universitaria), escala de Likert (muy de acuerdo, de acuerdo, ni de acuerdo ni en desacuerdo, en desacuerdo, muy en desacuerdo).

## Tabla Resumen variables

| Tipo de Variable    | Descripción                        | Ejemplos                                      |
|---------------------|------------------------------------|-----------------------------------------------|
| **Cuantitativa**     | Valores numéricos                  | Edad, peso, número de ventas                  |
| - **Discreta**       | Valores enteros                    | Número de hijos, cantidad de productos        |
| - **Continua**       | Cualquier valor en un intervalo    | Altura, temperatura                           |
| **Cualitativa**      | Categorías o cualidades            | Género, color de ojos, nivel de satisfacción  |
| - **Nominal**        | Sin orden natural                  | Color de cabello, país de origen              |
| - **Ordinal**        | Con orden natural                  | Nivel de estudios, escala de Likert           |


Vamos a extraer nuestros datos del *Titanic* desde el archivo .csv, crear un DataFrame de pandas y revisar los indicadores disponibles. Este conjunto de datos contiene información de 891 pasajeros con 12 atributos diferentes:

Variable                  | Descripción                                                                      | Tipo
--------------------------|----------------------------------------------------------------------------------|------------------------------
**Sobrevivió (Survived)** | Resultado de la supervivencia (0 = No; 1 = Sí)                                   | *Variable binaria*
**Clase (Pclass)**        | Clase socioeconómica (1 = Clase alta; 2 = Clase media; 3 = Clase baja)           | ________________
**Nombre (Name)**         | Nombre del pasajero                                                              | *Variable de texto*
**Sexo (Sex)**            | Sexo del pasajero                                                                | *Variable categórica nominal*
**Edad (Age)**            | Edad del pasajero                                                                | *Variable numérica discreta*
**SibSp**                 | Número de hermanos y cónyuges del pasajero a bordo                               | *Variable numérica discreta*
**Parch**                 | Número de padres e hijos del pasajero a bordo                                    | *Variable numérica discreta*
**Ticket**                | Número de boleto del pasajero                                                    | *Variable de texto*
**Tarifa (Fare)**         | Tarifa pagada por el pasajero                                                    | Variable numérica cuantitativa continua
**Cabina (Cabin)**        | Número de cabina del pasajero                                                    | *Variable de texto*
**Embarcado (Embarked)**  | Puerto de embarque del pasajero (C = Cherburgo; Q = Queenstown; S = Southampton) | ________________

**Ejercicio #1**

Completa el tipo de variable de **Embarcado**, **Tarifa** y **Clase**